kratos/python/add_search_strategies_to_python.cpp

```diff
@@ -124,6 +202,81 @@ void DefineSpecializedSpatialSearch(pybind11::module& m, const std::string& rCla
+    using ResultTypeContainer = SpatialSearchResultContainer<TObjectType>;
+    using ResultTypeContainerVector = SpatialSearchResultContainerVector<TObjectType>;
+    using SearchWrapperType = SearchWrapper<TSearchObject, TObjectType>;
+    using SearchWrapperPointerType = typename SearchWrapper<TSearchObject, TObjectType>::Pointer;
+
+    pybind11::class_<SearchWrapperType, SearchWrapperPointerType>(m, rClassName.c_str())
+    //.def(pybind11::init<NodesContainerType&, const DataCommunicator&>())
+    //.def(pybind11::init<NodesContainerType&, const DataCommunicator&, Parameters>())
+    .def(pybind11::init<ElementsContainerType&, const DataCommunicator&>())
+    .def(pybind11::init<ElementsContainerType&, const DataCommunicator&, Parameters>())
+    .def(pybind11::init<ConditionsContainerType&, const DataCommunicator&>())
+    .def(pybind11::init<ConditionsContainerType&, const DataCommunicator&, Parameters>())
+    .def("GetBoundingBox", &SearchWrapperType::GetBoundingBox)
+    .def("SearchInRadius", [&](SearchWrapperType& self, const Point& rPoint, const double Radius)  {
+        // Perform the search
+        auto p_results = Kratos::make_shared<ResultTypeContainer>();
+        self.SearchInRadius(rPoint, Radius, *p_results);
+        return p_results;
+    })
+    .def("SearchInRadius", [&](SearchWrapperType& self, const NodesContainerType& rNodes, const double Radius) {
+        // Perform the search
+        auto p_results = Kratos::make_shared<ResultTypeContainerVector>(); 
+        self.SearchInRadius(rNodes.begin(), rNodes.end(), Radius, *p_results);
+        return p_results;
+    })
+    .def("SearchNearestInRadius", [&](SearchWrapperType& self, const Point& rPoint, const double Radius)  {
+        // Perform the search
+        auto p_results = Kratos::make_shared<ResultTypeContainer>();
+        self.SearchNearestInRadius(rPoint, Radius, *p_results);
+        return p_results;
+    })
+    .def("SearchNearestInRadius", [&](SearchWrapperType& self, const NodesContainerType& rNodes, const double Radius) {
+        // Perform the search
+        auto p_results = Kratos::make_shared<ResultTypeContainerVector>(); 
+        self.SearchNearestInRadius(rNodes.begin(), rNodes.end(), Radius, *p_results);
+        return p_results;
+    })
+    .def("SearchNearest", [&](SearchWrapperType& self, const Point& rPoint) {
+        // Perform the search
+        auto p_results = Kratos::make_shared<ResultTypeContainer>();
+        self.SearchNearest(rPoint, *p_results);
+        return p_results;
+    })
+    .def("SearchNearest", [&](SearchWrapperType& self, const NodesContainerType& rNodes) {
+        // Perform the search
+        auto p_results = Kratos::make_shared<ResultTypeContainerVector>(); 
+        self.SearchNearest(rNodes.begin(), rNodes.end(), *p_results);
+        return p_results;
+    })
+    .def("SearchIsInside", [&](SearchWrapperType& self, const Point& rPoint) {
```

Comment:
what would this function mean if the object type is a node?

Reply:
a node is a type of point